### PR TITLE
ref(ts): Convert adminProjects to TypeScript

### PR DESCRIPTION
**Pull request by taylangocmen**

### CODE REVIEW

File: src/sentry/static/sentry/app/views/admin/adminProjects.tsx [hunk 1]
```diff
@@ -1,32 +1,54 @@
...
+  organization: Organization;
+  slug: string;
+  status: string;
+};
+
+type Props = {
+  location: Location;
+} & AsyncView['props'];
+
+export default class AdminProjects extends AsyncView<Props> {
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/admin/adminProjects.tsx:26]: Missing state right?

File: src/sentry/static/sentry/app/views/admin/adminProjects.tsx [hunk 2]
```diff
@@ -1,32 +1,54 @@
...
+  dateCreated: string;
+  name: string;
+  organization: Organization;
+  slug: string;
+  status: string;
+};
+
+type Props = {
+  location: Location;
+} & AsyncView['props'];
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/admin/adminProjects.tsx:24]: Might make more sense to use `RouteComponentProps`

File: src/sentry/static/sentry/app/views/admin/adminProjects.tsx (hunk 1 above)
Comment (taylangocmen) [reply src/sentry/static/sentry/app/views/admin/adminProjects.tsx:26]: We aren't using the state in this component is it necessary?

File: src/sentry/static/sentry/app/views/admin/adminProjects.tsx (hunk 2 above)
Comment (taylangocmen) [reply src/sentry/static/sentry/app/views/admin/adminProjects.tsx:24]: Makes sense

File: src/sentry/static/sentry/app/views/admin/adminProjects.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/admin/adminProjects.tsx]: Can we make it functional in that case?

File: src/sentry/static/sentry/app/views/admin/adminProjects.tsx [hunk 3]
```diff
@@ -1,32 +1,52 @@
...
 import moment from 'moment';
 
 import ResultGrid from 'app/components/resultGrid';
 import {t} from 'app/locale';
+import {Organization} from 'app/types';
+import AsyncView from 'app/views/asyncView';
 
-export const prettyDate = function (x) {
-  return moment(x).format('ll');
+const prettyDate = (x: string) => moment(x).format('ll');
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/admin/adminProjects.tsx:10]: maybe we should rename x to date?

File: src/sentry/static/sentry/app/views/admin/adminProjects.tsx [hunk 4]
```diff
@@ -1,32 +1,52 @@
...
+
+type Row = {
+  dateCreated: string;
+  name: string;
+  organization: Organization;
+  slug: string;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/admin/adminProjects.tsx:16]: is this a projectSlug? if yes, should we update it making things more clear?

File: src/sentry/static/sentry/app/views/admin/adminProjects.tsx (hunk 3 above)
Comment (taylangocmen) [reply src/sentry/static/sentry/app/views/admin/adminProjects.tsx:10]: It's used only once I'll move it inline, it'll be simpler

File: src/sentry/static/sentry/app/views/admin/adminProjects.tsx (hunk 4 above)
Comment (taylangocmen) [reply src/sentry/static/sentry/app/views/admin/adminProjects.tsx:16]: It is but it's clear with the context

Comment (taylangocmen) [reply src/sentry/static/sentry/app/views/admin/adminProjects.tsx:16]: It is also serialized as slug, not changing the name of the column in the api

File: src/sentry/static/sentry/app/views/admin/adminProjects.tsx [hunk 5]
```diff
@@ -1,32 +1,50 @@
...
+  status: string;
 };
 
-class AdminProjects extends React.Component {
-  getRow = row => [
-    <td>
+type Props = RouteComponentProps<{}, {}>;
+
+type State = {} & AsyncView['state'];
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/admin/adminProjects.tsx:20]:
```suggestion
type State = AsyncView['state'];
```

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**